src/OrchardCore.Modules/OrchardCore.Media/Services/MediaFileStoreResolverMiddleware.cs

```diff
@@ -14,7 +14,7 @@ namespace OrchardCore.Media.Services
     /// </summary>
     public class MediaFileStoreResolverMiddleware
     {
-        private static readonly ConcurrentDictionary<string, Lazy<Task>> Workers = new ConcurrentDictionary<string, Lazy<Task>>();
+        private readonly ConcurrentDictionary<string, Lazy<Task>> Workers = new ConcurrentDictionary<string, Lazy<Task>>();
```

Comment:
This here, where keys are paths, that i would have used `StringComparer.OrdinalIgnoreCase`

Reply:
Hmm, but here because related to concrete files (path are not only url path), maybe better to not use `OrdinalIgnoreCase` e.g if the file system is case sensitive.

But then afterwards you use `StartsWithNormalizedSegments` with `OrdinalIgnoreCase`.

Reply:
Okay the `OrdinalIgnoreCase` on `StartsWithNormalizedSegments` just acts on the `_assetsRequestPath`. Then this is the `_mediaFileStoreCache` that will return something or not depending on the case sensitivity of the file system.

**So finally i think it is good to use `OrdinalIgnoreCase` for the worker dictionary. This to prevent 2 concurrent works on the same file if the current file system is not case sensitive.**

Then if the file system is case sensitive, if you have 2 requests `/sitemap1.xml` and `/Sitemap1.xml`, even there are 2 different files they will be built one after the other, which is not a problem.